tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/PythonLanguageSpecificChecks.cs

```diff
@@ -75,4 +77,93 @@ public async Task<CLICheckResponse> AnalyzeDependenciesAsync(string packagePath,
             return new CookbookCLICheckResponse(0, $"Failed to run dependency analysis. Ensure tox is installed. Error: {ex.Message}", "https://docs.python.org/3/tutorial/venv.html");
         }
     }
-}
\ No newline at end of file
+
+    public async Task<CLICheckResponse> LintCodeAsync(string packagePath, CancellationToken ct = default)
+    {
+        try
+        {
+            _logger.LogInformation($"Starting pylint check for Python project at: {packagePath}");
+
+            // Find the repository root from the package path using GitHelper
+            var repoRoot = _gitHelper.DiscoverRepoRoot(packagePath);
+            _logger.LogInformation("Found repository root at: {RepoRoot}", repoRoot);
+
+            // Check if azure-sdk-tools is available, if not try to install it
+            var installed_tools = await EnsureAzureSdkToolsInstalledAsync(repoRoot, ct);
+
+            if (!installed_tools)
+            {
+                _logger.LogError("Required tools are not installed. Aborting pylint check.");
+                return new CLICheckResponse(1, "", "Required tools are not installed.");
+            }
+
+            // Run azpysdk pylint command
+            var command = "azpysdk";
+            var args = new[] { "pylint", packagePath };
```

Comment:
add --isolate tag so we create a venv for the check versus expecting someone to have it already